det3d/datasets/argoverse/argoverse_common.py

```diff
@@ -0,0 +1,330 @@
+import copy
+import json
+import os
+from typing import Dict, Tuple, List
+
+import numpy as np
+import torch
+from argoverse.data_loading.object_label_record import json_label_dict_to_obj_record
+from argoverse.data_loading.pose_loader import get_city_SE3_egovehicle_at_sensor_t
+from argoverse.utils import ply_loader, se3
+from scipy import interpolate
+from scipy.spatial.transform import Rotation, RotationSpline
+
+YSize = 80
+XSize = 144
+
+
+class Label:
+    """
+    Class to store the bounding boxes and label class of an occupancy grid
+    """
+
+    def __init__(self, record, label):
+        self.record = record
+        self.boxes = None
+        self.options = {"BACKGROUND": 0, "VEHICLE": 1}
+        self.label_class = label
+        self.label_num = self.options[self.label_class]
+
+
+def load_all_clouds(dataset: str, log_id: str, timestamp_list: List[int]) -> Dict:
+    """Loads all point clouds from the Argoverse Tracking set
+
+    Args:
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+        timestamp_list: (list[int]) List of timestamps to load point clouds from
+    Returns:
+        cloud_dict: (dict) A dictionary with the timestamp as key and point cloud as value
+    """
+    cloud_dict = {}
+    file_path = os.path.join(dataset, log_id, "lidar")
+    for timestamp in timestamp_list:
+        point_cloud = ply_loader.load_ply(
+            os.path.join(file_path, f"PC_{timestamp}.ply")
+        )
+        cloud_dict[timestamp] = point_cloud
+    return cloud_dict
+
+
+def perform_SE3(cloud_dict: Dict, dataset: str, log_id: str) -> Dict:
+    """Performs the SE3 transformation on all point clouds
+
+    Args:
+        cloud_dict: (dict) A dictionary with the timestamp as key and point cloud as value
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+    Returns:
+        cloud_dict: (dict) A dictionary with the timestamp as key and transformed point cloud as value
+    """
+    for k, v in cloud_dict.items():
+        SE3 = get_city_SE3_egovehicle_at_sensor_t(int(k), dataset, log_id)
+        cloud_dict[k] = SE3.transform_point_cloud(v)
+    return cloud_dict
+
+
+def grids_group_and_SE3(cloud_dict: Dict, dataset: str, log_id: str, n_t: int) -> Dict:
```

Comment:
do you mind adding the type annotation with the key and value types, like
`Dict[str,np.ndarray]`?